flake.nix

```diff
@@ -102,6 +119,47 @@
             done
             echo "All dependencies match"
           '';
+
+          test = with pkgs;
+            let
+              k-version =
+                lib.removeSuffix "\n" (builtins.readFile ./package/version);
+            in stdenv.mkDerivation {
+              name = "k-${k-version}-${self.rev or "dirty"}-test";
+              src = lib.cleanSource
+                (nix-gitignore.gitignoreSourcePure [ ./.gitignore ]
+                  ./k-distribution);
+              preferLocalBuild = true;
+              buildInputs = [ gmp mpfr k ];
+              postPatch = ''
+                patchShebangs tests/regression-new/*
+                substituteInPlace tests/regression-new/append/kparse-twice \
+                  --replace '"$(dirname "$0")/../../../bin/kparse"' '"${k}/bin/kparse"'
+                ${lib.optionalString stdenv.isDarwin ''
+                  for mak in include/kframework/*.mak
+                  do
+                    substituteInPlace $mak \
+                      --replace 'KOMPILE_FLAGS+=--no-exc-wrap' 'KOMPILE_FLAGS+=--no-exc-wrap --no-haskell-binary'
+                  done
+                ''}
```

Comment:
we add the `--no-haskell-binary` option to `KOMPILE_FLAGS` for all tests if running on macOS

Reply:
Please add a comment on why we added the `--no-haskell-binary` option.
We should try to add a fix eventually.
Maybe have an issue on the board.